File: algorithms/comparison_based_sorting/simple_comparison_based/SelectionSort/SelectionSort.go
package main

import "fmt"

// selectionSort sorts an array using the Selection Sort algorithm
func selectionSort(arr []int) {
    n := len(arr)
    // Traverse through all array elements
    for i := 0; i < n-1; i++ {
        // Find the minimum element in the unsorted portion of the array
        minIdx := i
        for j := i + 1; j < n; j++ {
            if arr[j] < arr[minIdx] {
                minIdx = j
            }
        }
        // Swap the found minimum element with the first element of the unsorted portion
        arr[i], arr[minIdx] = arr[minIdx], arr[i]
    }
}

func main() {
    // Example array
    arr := []int{64, 25, 12, 22, 11}
    // Sort the array using selectionSort
    selectionSort(arr)
    // Print the sorted array
    fmt.Println("Sorted array is:", arr)
}

File: algorithms/comparison_based_sorting/simple_comparison_based/SelectionSort/SelectionSort.py
def selection_sort(arr):
    n = len(arr)
    # Traverse through all array elements
    for i in range(n):
        # Find the minimum element in the unsorted portion of the array
        min_idx = i
        for j in range(i + 1, n):
            if arr[j] < arr[min_idx]:
                min_idx = j
        # Swap the found minimum element with the first element of the unsorted portion
        arr[i], arr[min_idx] = arr[min_idx], arr[i]

# Example array
arr = [64, 25, 12, 22, 11]
# Sort the array using selection_sort
selection_sort(arr)
# Print the sorted array
print("Sorted array is:", arr)